### ISSUE TITLE
Python 3.x crashes when you load a .so file in TCL through tkinter

### ISSUE THREAD

**Issue by vatsamail** [python/cpython#118051]
# Crash report

### What happened?

Python 3.x (3.9.7) is giving the following error if I try to load a shared object TCL file. If I run manually it works.

Manual method:

```
shell> tclsh
%load blah.so
%cmd_in_blah
```

Python method:

```
from tkinter import *
tcl = Tcl()
ret = tcl.eval("load blah.so")
```
Gives an error here >>>:

```
*** Error in `python': free(): invalid pointer: 0x00002ae034fc8d00 ***
Abort
```
Please help. Thanks in Advance.

### CPython versions tested on:

3.9

### Operating systems tested on:

Linux

**Comment by TheLizzard** [comment]
Can you provide `blah.so` or confirm that it happens with all other shared objects? Making the bug reproducible will help confirm and debug this problem.

**Comment by terryjreedy** [comment]
I don't know if loading `.so`s this way is supported, but I agree with TheLizzard's SO comment that aborting instead of raising is bad.

**Comment by vatsamail** [comment]
> Can you provide `blah.so` or confirm that it happens with all other shared objects? Making the bug reproducible will help confirm and debug this problem.
> 
> For the record, this bug comes from [this](https://stackoverflow.com/q/78345139/11106801) StackOverflow question.

Yes, I have a bunch of .so files that happen to work with tclsh 8.5 version when invoked standalone through load function but the same load functions errors out through tkinter. Also, I confirm that tkinter.TclVersion is 8.5

**Comment by terryjreedy** [comment]
Current tkinter requires 8.5.15, I believe, and only allows that because 1 known Linux distribution (out of about 20) lacked 8.6.  So if you are not on that distribution, please retry with up to date tkinter and tcl.  Use 3.12 if as all possible since everything before only gets security patches.

**Comment by vatsamail** [comment]
Tried to run the same with 3.12 Python. The same issue exists.

**Comment by serhiy-storchaka** [comment]
Could you please run it with a debugger and get a backtrace?

Does it crash in `tcl.eval()` or during Python shutdown? Try to run Python with the `-v` option and look whether it starts shutdown (prints "# crear ...") and what it output before crash.

**Comment by serhiy-storchaka** [comment]
I reproduced a crash of this kind, and in the most likely scenario this is not a CPython/`_tkinter` bug.

It crashes when the extension was built against a **different Tcl than Python links** and doesn't use Tcl's stubs mechanism, so `load` runs its `_Init` with the wrong ABI and corrupts memory (`free(): invalid pointer` / segfault). That would explain why it works in `tclsh` — the Tcl it was built against — but not under Python.

Fix is on the extension side: build it with stubs, or against the exact Tcl your Python uses. A stubs extension reports the mismatch cleanly instead of crashing.

Closing as not planned, since without the `.so` or a backtrace this can't be pinned down further — happy to reopen with either.
